### PR TITLE
:recycle: Refactor code

### PR DESCRIPTION
**Pull request by Quaquaro**

### CODE REVIEW

File: src/components/Dialog.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+Dialog.propTypes = {
+  onDialog: PropTypes.func,
+  message: PropTypes.string,
+  nameOfGame: PropTypes.string
+};
+
+export default function Dialog({ onDialog, message, nameOfGame }) {
+  return (
+    <Background onClick={() => onDialog(false)}>
+      <DialogBox onClick={(e) => e.stopPropagation()}>
```
Comment (JelenadlS) [inline src/components/Dialog.js:15]: Cool that you added it, its for clicking outside of the dialog and the window disappears, right?
I got crazy with it and left it away in the end😅

File: src/components/Dialog.spec.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    render(
+      <Theme>
+        <Dialog message="Tolle Sache" nameOfGame="Game" />
+      </Theme>
+    );
+
+    expect(screen.getAllByRole('button').length).toBe(2);
+    expect(screen.getAllByRole('heading').length).toBe(2);
+  });
+});
```
Comment (JelenadlS) [inline src/components/Dialog.spec.js:16]: Maybe you could also test the both onClicks here 😊

File: src/App.js [hunk 3]
```diff
@@ -18,7 +18,11 @@ function App() {
           <Route
             path="/"
             element={
-              games.length === 0 ? <GamesPage games={infoText} /> : <GamesPage games={games} />
+              games.length === 0 ? (
+                <GamesPage games={infoText} onDeleteGame={handleDeleteGame} />
```
Comment (JelenadlS) [inline src/App.js]: not sure if you need to have the delete function in it when only a text is visible without having saved something by myself or is it this example thing which is displaying and the user can therefore delete it?

**Review by JelenadlS — APPROVED** [review]
I think so far its looking good and also good structured, should also do it at one point haha😅
I just added one comment for testing the onClicks maybe as well - up to you 😊 and another one for the deleting part, however, might be that i just got it wrong 😊
By the way, i love that you put the icons into your commit messages🤩

File: src/components/Dialog.js (hunk 1 above)
Comment (Quaquaro) [reply src/components/Dialog.js:15]: Yes it is.

File: src/App.js (hunk 3 above)
Comment (Quaquaro) [reply src/App.js]: Im going to test it without the props, but i thought the GamesPage component need the probs anywere!

**Pull request merged**